superset-frontend/src/dashboard/components/SliceAdder.jsx

```diff
@@ -258,6 +295,29 @@ class SliceAdder extends React.Component {
     );
   }
 
+  onShowOnlyMyCharts(showOnlyMyCharts) {
+    if (!showOnlyMyCharts) {
+      this.slicesRequest = this.props.fetchSlices(
+        undefined,
+        this.state.searchTerm,
+        this.state.sortBy,
+      );
+    }
+    this.setState(prevState => ({
+      showOnlyMyCharts,
+      filteredSlices: this.getFilteredSortedSlices(
+        this.props.slices,
+        prevState.searchTerm,
+        prevState.sortBy,
+        showOnlyMyCharts,
+      ),
+    }));
+    setItem(
```

Comment:
it won't store the value when `showOnlyMyCharts` is true

Reply:
Did you mean, the first time? If that's the case, it's not a problem because the constructor defaults to `true`.

```
showOnlyMyCharts: getItem(
  LocalStorageKeys.dashboard__editor_show_only_my_charts,
  true,
),
```

Reply:
oh my mistake. I understood this operation under `if (!showOnlyMyCharts) { ... }` but it's not. it looks good